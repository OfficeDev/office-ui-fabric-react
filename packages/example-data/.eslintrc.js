// @ts-check
module.exports = {
  extends: ['../../scripts/eslint/v7'],
  root: true,
  rules: {
    'no-alert': 'off',
  },
};
