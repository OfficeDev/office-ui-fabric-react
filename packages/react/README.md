# @uifabric/react

**React components for [Office UI Fabric React](https://dev.microsoft.com/fabric)**

These are not production-ready components and **should never be used in product**. This space is useful for testing new components whose APIs might change before final release.

To import React components:

```js
import { ComponentName } from '@uifabric/react';
```

Once the React component graduates to a production release, the component will be available at:

```js
import { ComponentName } from 'office-ui-fabric-react';
```
