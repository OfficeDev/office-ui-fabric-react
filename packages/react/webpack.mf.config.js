const path = require('path');
const fs = require('fs');
const webpack = require('webpack');
const getResolveAlias = require('@fluentui/scripts/webpack/getResolveAlias');
const { createServeConfig } = require('@fluentui/scripts/webpack/webpack-resources');
const { webpackMerge } = require('just-scripts');

const shared = {
  react: { singleton: true },
  'react-dom': { singleton: true },
};

const rootComponents = fs
  .readdirSync(path.join(__dirname, 'src'))
  .filter(filename => filename[0].toUpperCase() === filename[0])
  .map(filename => filename.replace(/\.tsx?/, ''));
const rootComponentsExposes = {};
for (const component of rootComponents) {
  rootComponentsExposes[`./lib/${component}`] = `@fluentui/react/src/${component}`;
}

// TODO: fork-ts-checker-webpack-plugin needs to be taught about the resolve alias -> pathmapping in order for the type checking to work
process.env.SKIP_TYPECHECK = true;

const myConfig = {
  output: {},
  entry: './empty-entrypoint.js',
  mode: 'development',
  devtool: 'cheap-module-source-map',
  optimization: {
    concatenateModules: false,
  },
  module: {
    rules: [],
  },
  plugins: [
    new webpack.container.ModuleFederationPlugin({
      name: 'fluentuiReact',
      filename: 'remoteEntry.js',
      exposes: {
        ...rootComponentsExposes,
        './lib/compat/Button': '@fluentui/react/src/compat/Button',
        '.': '@fluentui/react/src/index',
        './lib/index': '@fluentui/react/src/index',
      },
      shared,
    }),
  ],
  resolve: {
    alias: getResolveAlias(),
  },
  devServer: {
    port: 2345,
    static: './dist',
  },
};

const metaConfig = webpackMerge.merge(createServeConfig(), myConfig);

module.exports = metaConfig;
