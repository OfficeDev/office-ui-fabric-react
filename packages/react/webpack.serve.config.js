const path = require('path');
const resources = require('@uifabric/build/webpack/webpack-resources');

module.exports = resources.createServeConfig({
  entry: './src/demo/index.tsx',

  output: {
    filename: 'demo-app.js',
  },

  externals: {
    react: 'React',
    'react-dom': 'ReactDOM',
  },

  resolve: {
    alias: {
      '@fluentui/react/src': path.join(__dirname, 'src'),
      '@fluentui/react/lib': path.join(__dirname, 'lib'),
      '@fluentui/react': path.join(__dirname, 'lib'),
      'Props.ts.js': 'Props',
      'Example.tsx.js': 'Example',
    },
  },
});
