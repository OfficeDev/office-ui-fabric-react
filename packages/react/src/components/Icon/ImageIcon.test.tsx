import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { ImageIcon } from './index';
import { TestImages } from '@fluentui/example-data';

describe('ImageIcon', () => {
  it('renders ImageIcon correctly', () => {
    const component = renderer.create(<ImageIcon imageProps={{ src: TestImages.iconOne }} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders named ImageIcon correctly', () => {
    const component = renderer.create(<ImageIcon imageProps={{ src: TestImages.iconOne, alt: 'image one' }} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders ImageIcon correctly with image aria-label', () => {
    const component = renderer.create(
      <ImageIcon
        imageProps={{
          src: TestImages.iconOne,
          'aria-label': 'image one',
        }}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders ImageIcon correctly with container aria-label', () => {
    const component = renderer.create(<ImageIcon aria-label="image one" imageProps={{ src: TestImages.iconOne }} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders ImageIcon correctly with image title', () => {
    const component = renderer.create(<ImageIcon imageProps={{ src: TestImages.iconOne, title: '"image one' }} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders ImageIcon correctly with container aria-labelledby', () => {
    const component = renderer.create(<ImageIcon aria-labelledby="id" imageProps={{ src: TestImages.iconOne }} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
