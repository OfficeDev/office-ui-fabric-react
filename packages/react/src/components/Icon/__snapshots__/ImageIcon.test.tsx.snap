// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageIcon renders ImageIcon correctly 1`] = `
<div
  aria-hidden={true}
  className=
      ms-Icon
      ms-Icon-imageContainer
      {
        display: inline-block;
      }
      {
        overflow: hidden;
      }
>
  <div
    className=
        ms-Image
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          overflow: hidden;
        }
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <img
      alt=""
      className=
          ms-Image-image
          ms-Image-image--portrait
          is-notLoaded
          is-fadeIn
          {
            display: block;
            opacity: 0;
          }
      onError={[Function]}
      onLoad={[Function]}
      src="https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets/icon-one.png"
    />
  </div>
</div>
`;

exports[`ImageIcon renders ImageIcon correctly with container aria-label 1`] = `
<div
  className=
      ms-Icon
      ms-Icon-imageContainer
      {
        display: inline-block;
      }
      {
        overflow: hidden;
      }
>
  <div
    className=
        ms-Image
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          overflow: hidden;
        }
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <img
      alt="image one"
      className=
          ms-Image-image
          ms-Image-image--portrait
          is-notLoaded
          is-fadeIn
          {
            display: block;
            opacity: 0;
          }
      onError={[Function]}
      onLoad={[Function]}
      src="https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets/icon-one.png"
    />
  </div>
</div>
`;

exports[`ImageIcon renders ImageIcon correctly with container aria-labelledby 1`] = `
<div
  className=
      ms-Icon
      ms-Icon-imageContainer
      {
        display: inline-block;
      }
      {
        overflow: hidden;
      }
>
  <div
    className=
        ms-Image
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          overflow: hidden;
        }
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <img
      aria-labelledby="id"
      className=
          ms-Image-image
          ms-Image-image--portrait
          is-notLoaded
          is-fadeIn
          {
            display: block;
            opacity: 0;
          }
      onError={[Function]}
      onLoad={[Function]}
      src="https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets/icon-one.png"
    />
  </div>
</div>
`;

exports[`ImageIcon renders ImageIcon correctly with image aria-label 1`] = `
<div
  className=
      ms-Icon
      ms-Icon-imageContainer
      {
        display: inline-block;
      }
      {
        overflow: hidden;
      }
>
  <div
    className=
        ms-Image
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          overflow: hidden;
        }
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <img
      aria-label="image one"
      className=
          ms-Image-image
          ms-Image-image--portrait
          is-notLoaded
          is-fadeIn
          {
            display: block;
            opacity: 0;
          }
      onError={[Function]}
      onLoad={[Function]}
      src="https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets/icon-one.png"
    />
  </div>
</div>
`;

exports[`ImageIcon renders ImageIcon correctly with image title 1`] = `
<div
  className=
      ms-Icon
      ms-Icon-imageContainer
      {
        display: inline-block;
      }
      {
        overflow: hidden;
      }
>
  <div
    className=
        ms-Image
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          overflow: hidden;
        }
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <img
      className=
          ms-Image-image
          ms-Image-image--portrait
          is-notLoaded
          is-fadeIn
          {
            display: block;
            opacity: 0;
          }
      onError={[Function]}
      onLoad={[Function]}
      src="https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets/icon-one.png"
      title="\\"image one"
    />
  </div>
</div>
`;

exports[`ImageIcon renders named ImageIcon correctly 1`] = `
<div
  className=
      ms-Icon
      ms-Icon-imageContainer
      {
        display: inline-block;
      }
      {
        overflow: hidden;
      }
>
  <div
    className=
        ms-Image
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          overflow: hidden;
        }
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <img
      alt="image one"
      className=
          ms-Image-image
          ms-Image-image--portrait
          is-notLoaded
          is-fadeIn
          {
            display: block;
            opacity: 0;
          }
      onError={[Function]}
      onLoad={[Function]}
      src="https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets/icon-one.png"
    />
  </div>
</div>
`;
